addons/test_mail/tests/test_message_post.py

```diff
@@ -798,6 +800,39 @@ def test_message_post(self):
                 partner_ids=self.partner_portal.ids,
             )
 
+    @mute_logger('odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    @users('employee')
+    def test_message_post_using_controller(self):
+        """ Test mail_message_post of ThreadController (especially on archived record). """
+        # Use an archivable test record
+        test_record = self.env['mail.test.activity'].create({})
+        body_text = 'Log a note on record'
+        post_data = {
+            'body': body_text,
+            'message_type': 'comment',
+            'subtype_xmlid': 'mail.mt_comment',
+            'not_valid': 'test',
+        }
+        post_data_keys = set(post_data.keys())
+
+        with MockRequest(self.env), patch.object(self.registry['mail.thread'], 'message_post') as mock:
+            ThreadController().mail_message_post(test_record._name, test_record.id, post_data)
+            self.assertTrue(mock.called)
+            self.assertEqual({key: value for key, value in mock.call_args.kwargs.items() if key in post_data_keys}, {
+                'body': Markup(body_text),
```

Comment:
I'm guessing the markup is necessary for body_text because of this, and you want to assess that method full behavior

```python
 if "body" in post_data:
    post_data["body"] = Markup(post_data["body"])  # contains HTML such as @mentions
```

Same for the `not_valid`, you check _get_allowed_message_post_params

Can you just explain _- en passant -_ what is mock.call_args.kwargs in this case? This would be the `message_data`?